application/ledger_query_service/src/ethereum/queries.rs

```diff
@@ -26,6 +27,89 @@ pub struct EthereumTransactionQuery {
     transaction_data_length: Option<usize>,
 }
 
+#[derive(Debug, Clone, Serialize, Deserialize, Default)]
+pub struct EthereumTransactionLogQuery {
+    logs: Vec<Vec<H256>>,
+}
+
+impl EthereumTransactionLogQuery {
+    pub fn matches_block(&self, block: &EthereumBlock<EthereumTransaction>) -> bool {
```

Comment:
:question: Is this used anywhere? Why do we need two ways to check for logs?

Reply:
This query is inherently different to the old transaction query, because it only looks at the logs of the transaction receipt.

Reply:
This checks if a block contains a transaction which has emitted the topics included in the query. It allows for skipping blocks that don't need to be looked at.